packages/lesswrong/components/posts/PostsDay.jsx

```diff
@@ -21,38 +20,82 @@ const styles = theme => ({
     marginLeft: "23px",
     color: "rgba(0,0,0,0.5)",
   },
+  shortformGroup: {
+    marginTop: 20,
+  },
+  shortformTag: {
+    ...theme.typography.body2,
+    ...theme.typography.commentStyle,
+    color: theme.palette.grey[700],
+    marginBottom: 8,
+  },
 })
 
-class PostsDay extends PureComponent {
-
-  render() {
-    const { date, posts, classes, currentUser } = this.props;
-    const noPosts = posts.length === 0;
-    const { PostsItem2 } = Components
+const PostsDay = ({ date, posts, results: comments, totalCount, loadMore, hideIfEmpty, classes, currentUser }) => {
+  const noPosts = !posts || (posts.length === 0);
+  const noComments = !comments || (comments.length === 0);
+  const { PostsItem2, CommentsNode, LoadMore } = Components
 
-    return (
-      <div className={classes.root}>
-        <Typography variant="headline" className={classes.dayTitle}>
-          <Hidden xsDown implementation="css">
-            {date.format('dddd, MMMM Do YYYY')}
-          </Hidden>
-          <Hidden smUp implementation="css">
-            {date.format('ddd, MMM Do YYYY')}
-          </Hidden>
-        </Typography>
-        { noPosts ? (<div className={classes.noPosts}>No posts on {date.format('MMMM Do YYYY')}</div>) :
-          <div>
-            {posts.map((post, i) => <PostsItem2 key={post._id} post={post} currentUser={currentUser} index={i} />)}
-          </div>
-        }
-      </div>
-    );
+  // The most recent day is hidden if there are no posts on it, to avoid having
+  // an awkward empty partial day when it's close to midnight.
+  if (noPosts && noComments && hideIfEmpty) {
```

Comment:
What happens if the user is not a beta user and there are comments?

Reply:
Then you get an awkward empty partial day. But that's not that bad.